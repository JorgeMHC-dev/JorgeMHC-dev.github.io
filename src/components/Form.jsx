import React, { useState } from 'react';
import { Button, TextField, Grid } from '@mui/material';

const SumForm = () => {
  const [firstNum, setFirstNum] = useState(null);
  const [secondNum, setSecondNum] = useState(null);
  const [sumValue, setSumValue] = useState(null);

  const sumNumbers = (event) => {
    event.preventDefault();

    setSumValue(Number(firstNum) + Number(secondNum));
  };

  return (
    <Grid container spacing={2}>
      <Grid item xs={12}>
        <h1>Suma de valores</h1>
      </Grid>
      <Grid item xs={12}>
        <TextField
          aria-label="first number"
          placeholder="First number"
          defaultValue={firstNum}
          type="number"
          onChange={(e) => setFirstNum(e.target.value)}
        />
      </Grid>
      <Grid item xs={12}>
        <TextField
          aria-label="second number"
          placeholder="Second number"
          defaultValue={secondNum}
          type="number"
          onChange={(e) => setSecondNum(e.target.value)}
        />
      </Grid>
      <Grid item xs={12}>
        <Button onClick={sumNumbers} variant="contained">
          Add
        </Button>
      </Grid>
      <Grid item xs={12}>
        <p>{sumValue}</p>
      </Grid>
    </Grid>
  );
};

export { SumForm };
