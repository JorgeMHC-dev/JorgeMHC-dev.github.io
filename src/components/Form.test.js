import { Form } from './Form';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

describe('App tests', () => {
  it('should display header title', () => {
    render(<Form />);
    const Header = screen.getByRole('heading', { name: 'Suma de valores' });

    expect(Header).toBeInTheDocument();
  });

  it('should display the result of the addition', async () => {
    render(<Form />);
    const FirstInput = screen.getByLabelText('first number');
    const SecondInput = screen.getByLabelText('second number');

    userEvent.type(FirstInput, '3');
    userEvent.type(SecondInput, '2');
    userEvent.click(screen.getByText('Add'));

    expect(await screen.findByText('5')).toBeInTheDocument();
  });

  it('should validate input numbers', () => {
    render(<Form />);
    const FirstInput = screen.getByLabelText('first number');
    const SecondInput = screen.getByLabelText('second number');

    userEvent.type(FirstInput, '3');
    userEvent.type(SecondInput, 'A');
    userEvent.click(screen.getByText('Add'));

    expect(screen.getByText('You must only use numbers')).toBeInTheDocument();
  });
});
