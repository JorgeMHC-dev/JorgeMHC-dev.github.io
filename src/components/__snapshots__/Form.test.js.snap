// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests renders correctly 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h1>
        Suma de valores
      </h1>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          first number
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            placeholder="First number"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1ftyaf0"
            >
              <span>
                first number
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r1:"
          id=":r1:-label"
        >
          second number
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r1:"
            placeholder="Second number"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1ftyaf0"
            >
              <span>
                second number
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Add
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <p />
    </div>
  </div>
</div>
`;
